src/lin_diffEq_HHL.jl

```diff
@@ -32,52 +74,48 @@ function prepare_init_state(tspan::Tuple,x::Vector,h::Float64,g::Function)
   init_state
 end
 
-function Array_QuEuler(tspan::Tuple,h::Float64,g::Function)
-  N_t = round(2*(tspan[2] - tspan[1])/h + 3)
+function Array_QuDiff(tspan::Tuple,h::Float64,g::Function,alg::alg_lin_diffeq)
+  N_t = Int(round(2*(tspan[2] - tspan[1])/h + 3))
+  sz = size(g(1))
   I_mat = Matrix{Float64}(I, size(g(1)));
   A_ = I_mat;
-  zero_mat = zero(I_mat);
-  tmp_A = Array{Float64,2}(undef,size(g(1)))
+  # Generates First two rows
+  A_ = [A_ zeros(sz[1],sz[1]*(N_t));-1*(I_mat + h*g(tspan[1])) I_mat zeros(sz[1],sz[1]*(N_t-1))]
+  #Generates additional rows based on k - step
+  if alg.step > 1
+    A_ = [A_; Aval(QuAB2,2*h + tspan[1],h,g) zeros(sz[1],sz[1]*(N_t-2))]
+  end
 
-  for j = 2: N_t +1
-    A_ = [A_ zero_mat]
+  for i = 3:alg.step-1
+    tA_ = Aval(QuAB2,(i-1)*h + tspan[1],h,g)
+    tA_ = [zeros(sz[1],sz[1]*(i-2)) tA_]
+    A_ = [A_; tA_ zeros(sz[1],sz[1]*(N_t-i))]
```

Comment:
I'm wondering since `alg.step` is known before we start the calculation. A pre-allocated array `A_` can be used instead? allocating new arrays (which is concatenation here should be very slow)

Reply:
I have made some changes. Please let me know if they are alright.